api/src/utils/xlsx-utils/worksheet-utils.ts

```diff
@@ -499,27 +509,39 @@ export async function getCBMeasurementsFromWorksheet(
   critterBaseService: CritterbaseService,
   sheet = DEFAULT_XLSX_SHEET_NAME
 ): Promise<TsnMeasurementMap> {
-  const tsnMeasurements: TsnMeasurementMap = {};
   const rows = getWorksheetRowObjects(xlsxWorksheets[sheet]);
+  const tsns = rows.map((row) => String(row['ITIS_TSN'] ?? row['TSN'] ?? row['TAXON'] ?? row['SPECIES']));
+  return getCBMeasurementsFromTSN(tsns, critterBaseService);
+}
 
+/**
+ * Fetches measurement definitions from critterbase for a given list of TSNs and creates and returns a map with all data fetched
+ * Throws if a TSN does not return measurements.
+ *
+ * @param {string[]} tsns List of TSNs
+ * @param {CritterbaseService} critterBaseService Critterbase service
+ * @returns {*} Promise<TsnMeasurementMap>
+ */
+export async function getCBMeasurementsFromTSN(
+  tsns: string[],
+  critterBaseService: CritterbaseService
+): Promise<TsnMeasurementMap> {
+  const tsnMeasurements: TsnMeasurementMap = {};
   try {
-    for (const row of rows) {
-      const tsn = String(row['ITIS_TSN'] ?? row['TSN'] ?? row['TAXON'] ?? row['SPECIES']);
+    for (const tsn of tsns) {
       if (!tsnMeasurements[tsn]) {
-        // TODO: modify Critter Base to accept multiple TSN at once
         const measurements = await critterBaseService.getTaxonMeasurements(tsn);
         if (!measurements) {
-          // TODO: Any data for a TSN that has no measurements is invalid and should reject the uploaded CSV
+          throw `No measurements found for tsn: ${tsn}`;
```

Comment:
I would probably only throw an instance of `Error` rather than a string (even though this is technically OK), mostly because a caller that might catch this exception is most likely expecting an instance of `Error`.
```javascript
throw new Error(`No measurements found for tsn: ${tsn}`);
```

Reply:
+1